### PR TITLE
fix(stepper): focus bugs

### PR DESCRIPTION
**Pull request by mdt2**
⚠️ Rebased on #2288 

## Description

This PR fixes two focus bugs in the Stepper component:
- Active state for the Infield Buttons sticks around after the mouse click is released
- Invalid variant keyboard focus outline doesn't fit the component
- Infield Button isn't focusable within the Stepper component (aligning with implementations from SWC and React teams per Patrick's guidance)
- Invalid hover state border colors aren't mismatched (infield buttons were getting a different color than the text field)

And also adds kitchen sink style Chromatic coverage for all the variants of Stepper.

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

[In the docs site](https://pr-2358--spectrum-css.netlify.app/stepper) (check Spectrum and Express theme):
- [x] Infield buttons don't show their focus state when Stepper has focus/keyboard focus
- [x] Infield buttons show their focus state when they get focus/keyboard focus state within Stepper
- [x] Invalid Stepper keyboard focus outline looks the same as the keyboard focus outline on valid Stepper
- [ ] Invalid Stepper hover border is just one color (be extra sure to check Express on this one too)
- [ ] Infield Buttons aren't keyboard accessible in the Stepper (aligning with implementations from SWC and React teams per Patrick's guidance)

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [ ] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [ ] VRTs have been run and looked at.
- [ ] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [ ] I have updated relevant storybook stories and templates.
- [ ] I have tested these changes in Windows High Contrast mode.
- [ ] If my change impacts **other components**, I have tested to make sure they don't break.
- [ ] If my change impacts **documentation**, I have updated the documentation accordingly.
- [ ] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

**Review by pfulton — COMMENTED** [review]
Thanks for taking care of this one!

It looks like this is slightly different in Firefox vs. Chrome, which is maybe ok. I'm a bit more concerned about the missing (overlapping?) focus indicator border when the stepper buttons are focused.

Firefox:
Screenshot 2023-12-13 at 2 27 28 PM

Chrome:
Screenshot 2023-12-13 at 2 29 07 PM

Also, and I think we could do this as a separate/follow-up ticket: can we get representation in Chromatic for the focus states?

File: components/stepper/stories/stepper.stories.js [hunk 1]
```diff
@@ -94,10 +95,169 @@ export default {
...
+	hideStepper,
+	...args
+}) => {
+	return html`
+		<div>
+			${Template({
+				...args
+			})}
+
+			${isChromatic() ? chromaticKitchenSink(args) : null}
```
Comment (pfulton) [inline components/stepper/stories/stepper.stories.js:113]: 😍

**Review by pfulton — COMMENTED** [review]
Looking much better! Thank you.

Just one question regarding Storybook: if I flip the `focused` control to true, it doesn't appear to affect the component in any visual way. Should it be turning off and on the focus indicator?

**Comment by mdt2** [comment]
> Looking much better! Thank you.
> 
> Just one question regarding Storybook: if I flip the `focused` control to true, it doesn't appear to affect the component in any visual way. Should it be turning off and on the focus indicator?

Good catch! I broke it in the Chromatic Coverage commit by passing args manually when I didn't need to. Should be fixed now.

**Review by pfulton — APPROVED** [review]

**Pull request merged**